cloud/scope/powervs_cluster_test.go

```diff
@@ -1570,3 +1573,415 @@ func getPowerVSClusterScopeParams() PowerVSClusterScopeParams {
+
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		networkID, err := clusterScope.getNetwork()
+		g.Expect(err).To(BeNil())
+		g.Expect(networkID).To(BeNil())
+	})
+}
+
+func TestIsDHCPServerActive(t *testing.T) {
+	var (
+		mockPowerVS *mockP.MockPowerVS
+		mockCtrl    *gomock.Controller
+	)
+
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockPowerVS = mockP.NewMockPowerVS(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+
+	t.Run("When DHCPID is empty", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.IBMPowerVSCluster.Spec.DHCPServer.ID = ptr.To("")
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		isActive, err := clusterScope.isDHCPServerActive()
+		g.Expect(err).To(Equal(fmt.Errorf("DHCP ID is empty")))
+		g.Expect(isActive).To(BeFalse())
+	})
+	t.Run("When GetDHCPServer returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.PowerVSClientFactory = func() (powervs.PowerVS, error) {
+			dhcpServer := &models.DHCPServerDetail{ID: ptr.To("dhcpID"), Status: ptr.To(string(infrav1beta2.DHCPServerStateError))}
+			mockPowerVS.EXPECT().GetDHCPServer(gomock.Any()).Return(dhcpServer, nil)
+			return mockPowerVS, nil
+		}
+		clusterScope, _ := NewPowerVSClusterScope(clusterScopeParams)
+
+		isActive, err := clusterScope.isDHCPServerActive()
+		g.Expect(err).ToNot(BeNil())
+		g.Expect(isActive).To(BeFalse())
+	})
+}
+
+func TestCheckDHCPServerStatus(t *testing.T) {
+	testCases := []struct {
+		name           string
+		dhcpServer     models.DHCPServerDetail
+		expectedStatus bool
+	}{
+		{
+			name:           "test DHCP server state build",
+			dhcpServer:     models.DHCPServerDetail{ID: ptr.To("dhcpIDBuild"), Status: ptr.To(string(infrav1beta2.DHCPServerStateBuild))},
+			expectedStatus: false,
+		},
+		{
+			name:           "test DHCP server state active",
+			dhcpServer:     models.DHCPServerDetail{ID: ptr.To("dhcpIDActive"), Status: ptr.To(string(infrav1beta2.DHCPServerStateActive))},
+			expectedStatus: true,
+		},
+		{
+			name:           "test DHCP server state error",
+			dhcpServer:     models.DHCPServerDetail{ID: ptr.To("dhcpIDError"), Status: ptr.To(string(infrav1beta2.DHCPServerStateError))},
+			expectedStatus: false,
+		},
+		{
+			name:           "test DHCP server state invalid value",
+			dhcpServer:     models.DHCPServerDetail{ID: ptr.To("dhcpIDError"), Status: ptr.To("InvalidState")},
+			expectedStatus: false,
+		},
+	}
+	for _, tc := range testCases {
+		g := NewWithT(t)
+		clusterScope := PowerVSClusterScope{}
+		t.Run(tc.name, func(_ *testing.T) {
+			status, _ := clusterScope.checkDHCPServerStatus(tc.dhcpServer)
+			g.Expect(status).To(Equal(tc.expectedStatus))
+		})
+	}
+}
+
+func TestCreateDHCPServer(t *testing.T) {
```

Comment:
Can we cover all error cases here?

Reply:
Since it was covered as part of other test case, I ignored. Have added all the testcases in this function now